src/controllers/blocks/BlocksController.ts

```diff
@@ -193,7 +202,10 @@ export default class BlocksController extends AbstractController<BlocksService>
 		const historicApi = await this.api.at(hash);
 
 		// We set the last param to true because we haven't queried the finalizedHead
-		BlocksController.sanitizedSend(res, await this.service.fetchBlock(hash, historicApi, options));
+		BlocksController.sanitizedSend(
+			res,
+			await this.service.fetchBlock(hash, historicApi, options, decodedXcmMsgsArg, paraId as string),
```

Comment:
1. `decodedXcmMsgsArg`, and `paraId` should be apart of the options arg. Options should be the last arg in the call. 

2. There needs to be some validation for paraId as well. This will ensure we don't need to typecast as well.

Reply:
1. Got it! Just to note that I didn't do it before because I was not sure if we wanted to affect the rest of the endpoints (where `options` is present). I added them now with this [commit](https://github.com/paritytech/substrate-api-sidecar/pull/1364/commits/f4211cbd343b6980ffd69e564f850cf2198d03ce) and adjusted the  `options` arg in the other endpoints too.